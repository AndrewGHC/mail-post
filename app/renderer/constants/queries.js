/* eslint-disable import/prefer-default-export */
import gql from 'graphql-tag';

export const QueryGetAllLists = gql`
  query GetAllLists {
    lists {
      id
      name
      total_subscribers
    }
  }
`;
