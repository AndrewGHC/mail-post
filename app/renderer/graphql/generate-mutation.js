import { QUERY_GET_LISTS } from '.';

export const generateImportCsv = (csvPath, name) => ({
  variables: {
    csvPath,
    name,
  },
  update: (store, { data: { importCsv } }) => {
    const data = store.readQuery({ query: QUERY_GET_LISTS });
    data.lists.push(importCsv);
    store.writeQuery({ query: QUERY_GET_LISTS, data });
  },
});

export const generateDeleteLists = ids => ({
  variables: {
    ids,
  },
  optimisticResponse: {
    deleteLists: ids.map(id => ({ id, __typename: 'List' })),
  },
  update: (store, { data: { deleteLists: listIds } }) => {
    const data = store.readQuery({ query: QUERY_GET_LISTS });
    const contains = val => listIds.some(({ id }) => id === val);
    const withoutIds = data.lists.filter(({ id }) => !contains(id));
    data.lists = withoutIds;
    store.writeQuery({ query: QUERY_GET_LISTS, data });
  },
});
