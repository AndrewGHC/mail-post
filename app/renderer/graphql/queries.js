/* eslint-disable import/prefer-default-export */
import gql from 'graphql-tag';

export const QUERY_GET_LISTS = gql`
  query GetAllLists {
    lists {
      id
      name
      total_subscribers
      createdAt
    }
  }
`;

export const QUERY_GET_CAMPAIGNS = gql`
  query GetAllCampaigns {
    campaigns {
      id
      name
      subject
      body
      listId
    }
  }
`;

export const QUERY_GET_SETTINGS = gql`
  query GetSettings {
    settings {
      amazonSESkey
      amazonSESSecretKey
      amazonRegion
      amazonWhiteLabelUrl
      amazonEmail
    }
  }
`;
