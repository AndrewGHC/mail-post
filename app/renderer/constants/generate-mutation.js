/* eslint-disable import/prefer-default-export */
import { QUERY_GET_LISTS } from './queries';

export const generateImportCsv = (csvPath, name) => ({
  variables: {
    csvPath,
    name,
  },
  update: (store, { data: { importCsv } }) => {
    const data = store.readQuery({ query: QUERY_GET_LISTS });
    data.lists.push(importCsv);
    store.writeQuery({ query: QUERY_GET_LISTS, data });
  },
});
