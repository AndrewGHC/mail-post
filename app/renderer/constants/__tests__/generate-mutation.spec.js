import td from 'testdouble';
import { generateImportCsv } from '../generate-mutation';
import { QUERY_GET_LISTS } from '../queries';

require('testdouble-jest')(td, jest);

describe('generate mutations', () => {
  let storeStub;
  beforeEach(() => {
    storeStub = {
      readQuery: td.function(),
      writeQuery: td.function(),
    };
  });

  describe('generateImportCsv', () => {
    const csvPath = 'a csv path';
    const name = 'a name';
    it('generateImportCsv snapshot', () => {
      expect(generateImportCsv(csvPath, name)).toMatchSnapshot();
    });

    it('generateImportCsv update', () => {
      const importCsv = generateImportCsv(csvPath, name);
      const data = {
        importCsv: 'test',
      };
      const cache = { lists: ['hi'] };
      const expectRead = { query: QUERY_GET_LISTS };
      const expectWrite = {
        ...expectRead,
        data: {
          lists: [...cache.lists, data.importCsv],
        },
      };
      td.when(storeStub.readQuery(expectRead)).thenReturn(cache);
      importCsv.update(storeStub, { data });
      td.verify(storeStub.writeQuery(expectWrite));
    });
  });
});

