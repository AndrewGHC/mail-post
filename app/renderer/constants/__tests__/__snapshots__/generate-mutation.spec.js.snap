// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate mutations generateImportCsv generateImportCsv snapshot 1`] = `
Object {
  "update": [Function],
  "variables": Object {
    "csvPath": "a csv path",
    "name": "a name",
  },
}
`;
